### PR TITLE
IGNITE-17086 Warnings added for long lazy SQL queries

### PR DESCRIPTION
**Pull request by oleg-vlsk**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/LongRunningQueryTest.java [hunk 1]
```diff
@@ -215,28 +315,117 @@ private void checkBigResultSet() throws Exception {
...
+
+            it.next();
+
+            ConcurrentHashMap qrys = GridTestUtils.getFieldValue(heavyQueriesTracker(), "qrys");
+
+            H2QueryInfo qry = (H2QueryInfo)qrys.keySet().iterator().next();
+
+            long extWait = GridTestUtils.getFieldValue(qry, "extWait");
+
+            assertTrue(extWait >= EXT_WAIT_TIME - EXT_WAIT_REL_ERR);
```
Comment (alex-plekhanov) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/LongRunningQueryTest.java:422]: Test still can be flaky in some cases, even with EXT_WAIT_REL_ERR. Root cause to this error is async updating of 
`U.currentTimeMillis()`. Instead I propose to modify `sleep` a little bit, we can replace `Thread.sleep(EXT_WAIT_TIME);` to something like:
```
            long sleepStartTs = U.currentTimeMillis();
            
            while (U.currentTimeMillis() - sleepStartTs <= EXT_WAIT_REL_ERR)
                doSleep(100L);
```
And in this case we can check extWait without  EXT_WAIT_REL_ERR

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridMapQueryExecutor.java [hunk 2]
```diff
@@ -862,14 +885,18 @@ else if (qryResults.cancelled())
...
 
-                        GridQueryNextPageResponse msg = prepareNextPage(
-                            nodeRess,
-                            node,
-                            qryResults,
-                            req.query(),
-                            req.segmentId(),
-                            req.pageSize(),
-                            dataPageScanEnabled);
+                        GridQueryNextPageResponse msg = h2.executeWithResumableTimeTracking(
```
[the remark this answers is not in the document]
Comment (oleg-vlsk) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridMapQueryExecutor.java:888]: Catch section added.

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2QueryInfo.java [hunk 3]
```diff
@@ -65,6 +74,9 @@ public class H2QueryInfo implements TrackableQuery {
     /** Query id. */
     private final long queryId;
 
+    /** Lock object. */
+    private final Object lock = new Object();
```
[the remark this answers is not in the document]
Comment (oleg-vlsk) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2QueryInfo.java]: Corrected.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/LongRunningQueryTest.java [hunk 4]
```diff
@@ -215,28 +315,117 @@ private void checkBigResultSet() throws Exception {
...
+
+            Thread.sleep(EXT_WAIT_TIME);
+
+            it.next();
+
+            ConcurrentHashMap qrys = GridTestUtils.getFieldValue(heavyQueriesTracker(), "qrys");
+
+            H2QueryInfo qry = (H2QueryInfo)qrys.keySet().iterator().next();
+
+            long extWait = GridTestUtils.getFieldValue(qry, "extWait");
```
[the remark this answers is not in the document]
Comment (oleg-vlsk) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/LongRunningQueryTest.java]: Getters introduced.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/LongRunningQueryTest.java (hunk 1 above)
Comment (oleg-vlsk) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/LongRunningQueryTest.java]: Done.

**Pull request closed without merging**